### DISCUSSION TITLE
Cytron Motion 2350 Pro COM port not appearing

### DISCUSSION THREAD

**Discussion by dennisma** [earlephilhower/arduino-pico#2453] [Q&A] [answered]
Using a Cytron Motion 2350 Pro boards and latest release 4.0.2

I can upload code using the UF2 method (hold bootselect and power on) but I'm not seeing the COM port appear after doing so like I do with the RP2040 versions.

Just curious if this has not been fully implemented yet.

Thanks!!

**Comment by Andy2No** [comment]
Have you tried unplugging the USB cable and plugging it back in again?

I've been having to do that with a Pi Pico (RP2040), lately, with a project I've been tinkering with (extending [TinyBasicPlus ](https://github.com/BleuLlama/TinyBasicPlus)a bit).  I can't say I understand why, but that makes the port appear again, and I have to do it after each upload.

**Reply by dennisma** [reply]
I tried many things.   I have many different RP2040's and have been using this long enough to know what normally causes the port to not appear.   For example setting an I2C pin to SDA when it can only do SCL will fault the board and make the COM port not appear.

In this case... I upload a blank program to the RP2350 and no COM port appears.  I can also upload a blink program and it works fine so I know I am not hardware faulting.

My only assumption is the RP2350/Cytron board is so new everything is not quite implemented yet.    I can live with it easily doing a UF2 boot but wanted to make sure it was not implemented yet.

**Reply by earlephilhower** [reply]
The Cytron was contributed by a user, but there's nothing special about the board as far as the core or USB goes (since the RP2350 SOC runs everything in SW).  It's probably all good, just a config thing.

If `blink.ino` runs and blinks the LED then the USB port is working, but Windows isn't happy for some reason.  Check the Zadig installation instructions on https://github.com/earlephilhower/arduino-pico/blob/master/docs/install.rst#windows-7-driver-notes .  Each board will give its own USB VID/PID per the configuration and Windows expects a special driver for each different one (unlike Mac or Linux where the OS goes "oh, you're a CDC serial port?  I'll automatically hook up a /dev/ttyxxxx for you").

**Reply by dquadros** [reply]
Windows 10 will use a default CDC driver if there is no special driver assigned to the VID/PID (but still assign a new COM port if it thinks it is a new device - I am up to COM 70). You can check if the board was detected and what driver (if any) was used by look in the Device Manager.

**Reply by earlephilhower** [reply]
> ...`COM70`...

And to think, I was upset that I have `/dev/ttyACM1` and `/dev/ttyACM2` on my system.  :laughing:

**Reply by dennisma** [reply, marked as answer]
> The Cytron was contributed by a user, but there's nothing special about the board as far as the core or USB goes (since the RP2350 SOC runs everything in SW). It's probably all good, just a config thing.
> 
> If `blink.ino` runs and blinks the LED then the USB port is working, but Windows isn't happy for some reason. Check the Zadig installation instructions on https://github.com/earlephilhower/arduino-pico/blob/master/docs/install.rst#windows-7-driver-notes . Each board will give its own USB VID/PID per the configuration and Windows expects a special driver for each different one (unlike Mac or Linux where the OS goes "oh, you're a CDC serial port? I'll automatically hook up a /dev/ttyxxxx for you").

It appears to be a hardware issue with the board.   I got another Cytron Motion 2350 board in and it works perfectly fine.   I'm going to see if I can figure this other one out but I'm happy at least one of them is working.

Thanks again!
